mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java

```diff
@@ -98,1014 +100,1053 @@
-        }
-
-        return true;
-    }
-
-    /**
-     * Returns a projection containing the requested projection, all identity
-     * fields, and the objectType field
-     */
-    private Projection getProjectionFields(Projection requestedProjection,
-                                           EntityMetadata md) {
-        Field[] identityFields = md.getEntitySchema().getIdentityFields();
-        List<Projection> projectFields = new ArrayList<>(identityFields == null ? 1 : identityFields.length + 1);
-        if (requestedProjection instanceof ProjectionList) {
-            projectFields.addAll(((ProjectionList) requestedProjection).getItems());
-        } else if (requestedProjection != null) {
-            projectFields.add(requestedProjection);
-        }
-        if (identityFields != null) {
-            for (Field x : identityFields) {
-                projectFields.add(new FieldProjection(x.getFullPath(), true, false));
-            }
-        }
-        projectFields.add(new FieldProjection(DocTranslator.OBJECT_TYPE, true, false));
-
-        return new ProjectionList(projectFields);
-    }
-
-    private Error analyzeException(Exception e, final String otherwise) {
-        return analyzeException(e, otherwise, null, false);
-    }
-
-    private Error analyzeException(Exception e, final String otherwise, final String msg, boolean specialHandling) {
-        if (e instanceof Error) {
-            return (Error) e;
-        }
-
-        if (e instanceof MongoException) {
-            MongoException me = (MongoException) e;
-            if (me.getCode() == 18) {
-                return Error.get(CrudConstants.ERR_AUTH_FAILED, e.getMessage());
-            } else if (me instanceof MongoTimeoutException
-                    || me instanceof MongoExecutionTimeoutException) {
-                LOGGER.error(CrudConstants.ERR_DATASOURCE_TIMEOUT, e);
-                return Error.get(CrudConstants.ERR_DATASOURCE_TIMEOUT, e.getMessage());
-            } else if (me instanceof DuplicateKeyException) {
-                return Error.get(MongoCrudConstants.ERR_DUPLICATE, e.getMessage());
-            } else if (me instanceof MongoSocketException) {
-                LOGGER.error(MongoCrudConstants.ERR_CONNECTION_ERROR, e);
-                return Error.get(MongoCrudConstants.ERR_CONNECTION_ERROR, e.getMessage());
-            } else {
-                LOGGER.error(MongoCrudConstants.ERR_MONGO_ERROR, e);
-                return Error.get(MongoCrudConstants.ERR_MONGO_ERROR, e.getMessage());
-            }
-        } else if (msg == null) {
-            return Error.get(otherwise, e.getMessage());
-        } else if (specialHandling) {
-            return Error.get(otherwise, msg, e);
-        } else {
-            return Error.get(otherwise, msg);
-        }
-    }
+	public static final String ID_STR = "_id";
```

Comment:
Please undo the formatting changes so we can review. It also makes it hard to understand who modified a piece of code, because if we git blame on this file it will say all you because you reformatted, but you didn't really modify the logic. 

It looks like you are using tabs but the code in this project uses spaces. I would keep using spaces instead of tabs (pretty much all of our projects use spaces instead of tabs).